src/applications/caregivers/components/PreSubmitInfto/components/SignatureInput/index.jsx

```diff
@@ -0,0 +1,56 @@
+import React, { useEffect, useRef, useState } from 'react';
+import ErrorableTextInput from '@department-of-veterans-affairs/formation-react/ErrorableTextInput';
+
+const SignatureInput = ({ fullName, required, label, setIsSigned }) => {
+  const inputRef = useRef(null);
+  const [signature, setSignature] = useState({
+    value: '',
+    dirty: false,
+  });
+
+  const [hasError, setError] = useState(false);
+  const firstAndLast =
+    fullName?.first && fullName?.last && `${fullName.first} ${fullName.last}`;
+  const signatureMatchesFullName = firstAndLast === signature.value;
+
+  useEffect(
+    () => {
+      const isDirty = signature.dirty;
+      const activeElementId = document.activeElement.id;
+      const inputId = inputRef.current.inputId;
+      setIsSigned(signatureMatchesFullName);
+
+      if (isDirty && activeElementId !== inputId && !signatureMatchesFullName) {
+        setError(true);
+      }
+
+      if (signatureMatchesFullName) {
+        setError(false);
+      }
+    },
+    [setIsSigned, signature, signatureMatchesFullName],
+  );
+
+  useEffect(() => {}, []);
+  return (
+    <div>
+      <label htmlFor="vet-signature-input">
+        {label}
+        's full name
```

Comment:
We may want to change `'` to `&apos;`

Reply:
I don't mind it either way, just wondering what the preference is, i've used those codes before for symbols but curious to what the diff is with the apostrophe.

Reply:
TLDR Accessibility 
https://www.joelonsoftware.com/2003/10/08/the-absolute-minimum-every-software-developer-absolutely-positively-must-know-about-unicode-and-character-sets-no-excuses/

Reply:
Good to know, thanks Aaron!